packages/trigger-sdk/src/triggerClient.ts

```diff
@@ -1322,6 +1371,38 @@ export class TriggerClient {
       account: execution.account,
       source,
     };
+
+    if (execution.batched) {
+      if (!execution.context) {
+        throw new Error("Context needs to be defined for batched events.");
+      }
+
+      const parsedContext = z.array(z.any()).parse(JSON.parse(execution.context));
+
+      if (parsedContext.length !== execution.eventIds.length) {
+        throw new Error("Context length needs to match number of event IDs.");
+      }
+
+      return {
+        ...sharedTriggerContext,
+        events: execution.eventIds.map((id, i) => ({
+          id,
+          name: execution.event.name,
+          context: parsedContext[i],
+          timestamp: execution.event.timestamp,
```

Comment:
Cheating here - timestamp comes from the first event. Timestamps would either be another thing to attach to JobRun, get for each execution, or remove.

Alternatively, we could create [many-to-many relations](https://www.prisma.io/docs/orm/prisma-schema/data-model/relations/many-to-many-relations#relational-databases) between JobRun and EventRecord. This would be a big change.